esp/esdllib/README.md

```diff
@@ -0,0 +1,2142 @@
+    <LoggingManager name="my_service_logman">
+      ...LogManager configuration...
+      <LogAgent name="my_logagent" type="" services="">
+        ...LogAgent configuration...
+        <XSL>
+          <Transform>
+            <![CDATA[<xsl:stylesheet>...XSLT...</xsl:stylesheet>]]>
+          </Transform>
+        </XSL>
+      </LogAgent>
+      ...additional LogAgents...
+    </LoggingManager>
+  </Definition>
+</Binding>
+```
+
+Each `<Scripts>` element contains ESDL Script entry points that are executed as the transaction proceeds through its lifetime. Scripts in the `<Methods>` section are referred to as *service-level scripts*. They are executed by each method in a service. Scripts in each `<Method>` section are referred to as *method-level scripts* and are only executed when that method is run.
+
+As the ESP processes a Method request and reaches each script entry point, it executes all of that entry point's scripts. The service-level scripts execute first, in order, followed by the method-level scripts in order. The CDATA sections ensure that the order of operations within each entry point's script is preserved.
+
+ESDL service methods must have a `<Method>` element in the *service bundle config* to be invoked. The `@auth_feature` attribute on the `<Definition>` or `<Method>` elements specify security as described in the [Security Readme](https://github.com/hpcc-systems/HPCC-Platform/master/esp/esdllib/README-SECURITY.md#esdl-binding).
+
+The `<LoggingManager>` section contains settings for the service's logging. The `<LogAgent>` elements may contain `XSL/Transform` elements used to customize the outgoing logging request before it is dispatched.
+
+It is easiest to manage the ESDL Scripts, ESDL definitions and XSLT transforms that make up the *service bundle configuration* as separate files under source control. An upcoming feature of the `esdl` tool is a command to compile a *service bundle configuration* from these separate files.
+
+### Application Mode
+
+The Application Mode ESP deployment is designed for containerized cloud deployments, but can also work on bare metal. Part of what makes it so well suited to cloud deployments is that the ESP loads all configuration from files. An ESP can be launched as one of several different kinds of applications including `eclwatch` and `esdl`. We will only consider the *esdl application* here.
+
+An *esdl application* configuration starts with an `esp.yaml` *process configuration* file, though in cloud deployments it would be called a *pod configuration*. Each application mode has reasonable default values for all settings, so your file need only include customized settings. Refer to the default application mode configuration files in the [HPCCSystems repository](https://github.com/hpcc-systems/HPCC-Platform/tree/master/esp/applications/esdl) for some examples. Note that the cloud startup procedure includes a step that rolls up all the separate yaml files into `esp.yaml` prior to launch.
```

Comment:
Cloud vs kubernetes gets a little confusing here.  but I would say:

"it would be called a *configmap*" for the pod and would usually be generated by the hpcc helm installation.

Reply:
" cloud startup procedure" ==> "helm installation procedure".

Reply:
fixed